plenoptic/simulate/models/portilla_simoncelli.py

```diff
@@ -0,0 +1,1072 @@
+                    0 : self.n_orientations, 0:nrp, this_scale
+                ] = self.compute_crosscorrelation(
+                    orientation_bands_real.t(), next_scale_real, band_num_el
+                )
+                if (
+                    this_scale == self.n_scales - 1
+                ):  # correlations on the low-pass residuals
+                    self.representation["cross_orientation_correlation_real"][
+                        0:nrp, 0:nrp, this_scale + 1
+                    ] = self.compute_crosscorrelation(
+                        next_scale_real.t(), next_scale_real, (band_num_el / 4.0)
+                    )
+
+    def compute_crosscorrelation(self, ch1, ch2, band_num_el):
+        r"""Computes either the covariance of the two matrices or the cross-correlation
+        depending on the value self.use_true_correlations.
+
+        Parameters
+        ----------
+        ch1: torch.Tensor
+            First matrix for cross correlation.
+        ch2: torch.Tensor
+            Second matrix for cross correlation.
+        band_num_el: int
+            Number of elements for bands in the scale
+
+        Returns
+        =======
+
+        --: torch.Tensor
+            cross-correlation.
+
+        """
+
+        if self.use_true_correlations:
+            return ch1 @ ch2 / (band_num_el * ch1.std() * ch2.std())
+        else:
+            return ch1 @ ch2 / (band_num_el)
+
+    def compute_autocorrelation(self, ch):
+        r"""Computes the autocorrelation and variance of a given matrix (ch)
+
+        Parameters
+        ----------
+        ch: torch.Tensor
+
+
+        Returns
+        =======
+        ac: torch.Tensor
+            Autocorrelation of matrix (ch).
+
+        vari: torch.Tensor
+            Variance of matrix (ch).
+
+        """
+
+        channel_size = torch.min(torch.tensor(ch.shape[-2:])).to(float)
+
+        # Calculate the edges of the central auto-correlation
+        center = int(np.floor([(self.spatial_corr_width - 1) / 2]))
+        le = int(np.min((channel_size / 2.0 - 1, center)))  # center of the image ???
+
+        # Find the center of the channel
+        cy = int(ch.shape[-1] / 2)
+        cx = int(ch.shape[-2] / 2)
+
+        # Calculate the auto-correlation
+        ac = torch.fft.fft2(ch.squeeze())
+        ac = ac.real.pow(2) + ac.imag.pow(2)
+        ac = torch.fft.ifft2(ac)
+        ac = torch.fft.fftshift(ac.unsqueeze(0)).squeeze() / torch.numel(ch)
+
+        # Return only the central auto-correlation
+        ac = ac.real[cx - le : cx + le + 1, cy - le : cy + le + 1]
+        vari = ac[le, le]
+
+        if self.use_true_correlations:
+            ac = ac / vari
+
+        return ac, vari
+
+    def compute_skew_kurtosis(self, ch, vari):
+        r"""Computes the skew and kurtosis of ch given the ratio of its
+        variance (vari) and the pixel variance of the original image are
+        above a certain threshold.  If the ratio does not meet that threshold
```

Comment:
I can't give a confident/ meaningful answer to this question that I feel comfortable adding into the comments.  That variance lives in the denominator of the skew calculation so if it is very very small it would blow up the skew.  I'm guessing that would lead to unstable synthesis.